behavior/plans/plans.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import json
+import logging
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from pandas import DataFrame
+
+
+class PlanTree:
+    def __init__(self, query_id: str, json_plan: dict[str, Any]):
+        self.query_id: str = query_id
+        self.root: PlanNode = PlanNode(json_plan)
+        self.parent_id_to_child_ids: dict[int, list[int]] = build_id_map(self.root)
+
+
+class PlanNode:
+    def __init__(self, json_plan: dict[str, Any]):
+        self.node_type: str = json_plan["Node Type"]
+        self.startup_cost: float = json_plan["Startup Cost"]
+        self.total_cost: float = json_plan["Total Cost"]
+        self.plan_node_id: int = json_plan["plan_node_id"]
+        self.depth: int = json_plan["depth"]
+        self.plans: list[PlanNode] = (
+            [PlanNode(child_plan) for child_plan in json_plan["Plans"]]
+            if "Plans" in json_plan
+            else []
+        )
+
+    def __repr__(self) -> str:
+        indent = ">" * (self.depth + 1)  # incremented so we always prefix with >
+        return f"{indent} type: {self.node_type}, total_cost: {self.total_cost}, node_id: {self.plan_node_id}, depth: {self.depth}"
+
+
+def build_id_map(root: PlanNode) -> dict[int, list[int]]:
+    node_map: dict[int, list[int]] = {}
+    node_map[root.plan_node_id] = [child.plan_node_id for child in root.plans]
+
+    for child in root.plans:
+        _build_id_map(child, node_map)
+
+    return node_map
+
+
+def _build_id_map(node: PlanNode, node_map: dict[int, list[int]]) -> None:
+    node_map[node.plan_node_id] = [child.plan_node_id for child in node.plans]
+    for child in node.plans:
+        _build_id_map(child, node_map)
+
+
+def show_plan_tree(plan_tree: PlanTree) -> None:
+    logging.info("\n===== QueryID: %s =====", plan_tree.query_id)
+    logging.info("Parent ID to Child IDs: %s", plan_tree.parent_id_to_child_ids)
+    logging.info("%s", plan_tree.root)
+
+    for child in plan_tree.root.plans:
+        show_plan_node(child)
+
+
+def show_plan_node(plan_node: PlanNode) -> None:
+    logging.info("%s", plan_node)
+
+    for child in plan_node.plans:
+        show_plan_node(child)
+
+
+def set_node_ids(json_plan: dict[str, Any]) -> None:
+    json_plan["plan_node_id"] = 0
+    json_plan["depth"] = 0
+    next_node_id: int = 1
+
+    if "Plans" in json_plan:
+        for child in json_plan["Plans"]:
+            next_node_id = _set_node_ids(child, next_node_id, 1)
+
+
+def _set_node_ids(json_plan: dict[str, Any], next_node_id: int, depth: int) -> int:
```

Comment:
If it's okay with you I'm going to remove this file from the PR because it's not currently called or needed.  Matt's addition of left/right child plan node identifiers removed the need for this.  This code may be useful down the road for parsing explain plans from pg_store_plans or otherwise, so I can keep it on a branch of my fork.